### PR TITLE
cache result for getowner

### PR DESCRIPTION
**Pull request by icewind1991**
significant performance increase when working with large directories

~5% improvement in my setup with ~1000 entries in the directory

[comparison](https://blackfire.io/profiles/compare/c4b470f1-2460-4373-9f4a-3fba96488854/graph)

cc @PVince81

### CODE REVIEW

File: lib/private/files/storage/common.php [hunk 1]
```diff
@@ -383,7 +383,13 @@ public function getStorageCache($storage = null) {
 	 * @return string|false uid or false
 	 */
 	public function getOwner($path) {
-		return \OC_User::getUser();
+		static $owner;
+
+		if (!isset($owner)) {
+			$owner = \OC_User::getUser();;
```
Comment (LukasReschke) [inline lib/private/files/storage/common.php]: Double ;

**Comment by karlitschek** [comment]
nice :+1:

File: lib/private/files/storage/common.php [hunk 2]
```diff
@@ -383,7 +383,13 @@ public function getStorageCache($storage = null) {
 	 * @return string|false uid or false
 	 */
 	public function getOwner($path) {
-		return \OC_User::getUser();
+		static $owner;
```
Comment (nickvergessen) [inline lib/private/files/storage/common.php]: this f...s up unit tests. The object is not static, so please use a member variable?

**Comment by PVince81** [comment]
Is `getUser()` really expensive ? Then there might be other places in the code that could benefit optimizing.

:+1:

**Comment by icewind1991** [comment]
This is the only place where we call it often afaik

**Pull request merged**